File: Java/LongestCommonPrefix.java
//
//  LeetCode
//  Created by Abdelrahman Nasser
//

class Solution {
    public String longestCommonPrefix(String[] strs) {
       if (strs.length == 0)
            return "";
        Arrays.sort(strs);
        String word = strs[0];
        String prefix = "";
        
        for (int i = 0; i < word.length(); i ++) {
            prefix = word.substring(0, i+1);
            int len = prefix.length();
            
            for (int j = 1; j < strs.length; j ++) {
                
                if (strs[j].substring(0, len).equals(prefix)) {
                    continue;
                } else {
                    return prefix.substring(0, i);
                }
            }
        }
        
        return prefix; 
    }
}

File: Swift/LongestCommonPrefix.swift
//
//  LeetCode
//  Created by Abdelrahman Nasser
//

class Solution {
    func longestCommonPrefix(_ strs: [String]) -> String {
        if(strs.count == 0) {
           return "";
       }

       var output = strs[0];

       for i in 1..<strs.count {
           output = findCommonPrefix(output, strs[i]);
       }

       return output;
    }

    func findCommonPrefix(_ str1: String, _ str2: String) -> String {
        let chars1 = Array(str1)
        let chars2 = Array(str2)
        var output = "";

        var i = 0;

        while(i < chars1.count && i < chars2.count && chars1[i] == chars2[i]) {
            output = output + String(chars1[i]);
            i += 1;
        }

        return output;
    }
}
